packages/core/src/Agentica.ts

```diff
@@ -17,11 +17,27 @@ import { IAgenticaProps } from "./structures/IAgenticaProps";
 import { IAgenticaProvider } from "./structures/IAgenticaProvider";
 import { IAgenticaTokenUsage } from "./structures/IAgenticaTokenUsage";
 
+type Middleware = (
+  /**
+   * The context object for the current request, containing relevant data
+   * that middleware functions can read or modify.
+   */
+  ctx: IAgenticaContext,
+
+  /**
+   * A function that passes control to the next middleware in the chain.
+   *
+   * - Must be called to continue execution; otherwise, the chain will stop.
+   * - Returns a `Promise<void>`, so it should be awaited (`await next()`).
+   * - If not called, the current middleware will act as a termination point.
+   */
+  next: () => Promise<void>,
+) => any;
```

Comment:
I think this return value should be a `Awaitable<void>`
if return type is `any`, it means that it is being used somewhere